mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb

```diff
@@ -781,10 +836,19 @@ func resourceComputeInstanceGroupManagerUpdate(d *schema.ResourceData, meta inte
 	}
 	<% end -%>
 
-	if d.HasChange("stateful_disk") {
-		updatedManager.StatefulPolicy = expandStatefulPolicy(d.Get("stateful_disk").(*schema.Set).List())
+	isStatefulPolicyExpanded := false
+	//Add validation for isStatefulPolicyExpanded here to avoid variable declared but not used in GA
+	if d.HasChange("stateful_disk") && !isStatefulPolicyExpanded {
+		updatedManager.StatefulPolicy = expandStatefulPolicy(d)
 		change = true
+		isStatefulPolicyExpanded = true
 	}
+	<% unless version == "ga" -%>
+	if d.HasChange("stateful_internal_ips") || d.HasChange("stateful_external_ips") && !isStatefulPolicyExpanded {
```

Comment:
it would great to simplify these - for example:
```
<% unless version == "ga" -%>
	if d.HasChange("stateful_disk") || d.HasChange("stateful_internal_ips") || d.HasChange("stateful_external_ips") {
<% else -%>
        if d.HasChange("stateful_disk") {
<% end -%>
```

That would let us just have one block for expanding this field.